lib/govuk_navigation_helpers/navigation_sidebar.rb

```diff
@@ -0,0 +1,85 @@
+require 'govuk_navigation_helpers/services'
+require 'govuk_navigation_helpers/configuration'
+
+module GovukNavigationHelpers
+  class NavigationSidebar
+    def initialize(content_item)
+      @content_item = ContentItem.new content_item
+    end
+
+    def navigation_sidebar
+      {
+        related_items: related_items,
+        collections: related_collections,
+        topics: related_topics,
+        policies: related_policies,
+        publishers: related_organisations,
+        other: [related_external_links, related_contacts]
+      }
+    end
+
+  private
+
+    def process_links_for_sidebar(related_content)
+      related_content.map do |link|
```

Comment:
It seems like we `map` a collection of links regularly, with the variation that the path key is sometimes `"base_path"` and sometimes `"url"`. This suggests there's a possibly way to satisfy all the cases here and simplify some of the link hash building/processing. Might DRY the code up by a few lines.
Maybe
```ruby
def build_links_for_sidebar(collection, path_key="base_path", additional_attrs={})
```
`additional_attrs` could contain `rel: "external"` where necessary and be merged into the result.
Can discuss if you like.